src/App.jsx

```diff
@@ -39,7 +43,23 @@ const mockedAuthorsList = [
 function App() {
 	const [coursesList, setCoursesList] = useState(mockedCoursesList);
 	const [authorsList, setAuthorsList] = useState(mockedAuthorsList);
-	const [step, setStep] = useState('list');
+	const navigate = useNavigate();
+	const pathname = useLocation().pathname;
+
+	// Redirecting to the login page if the user is not authorized
+	useEffect(() => {
+		let userToken = localStorage.getItem('user');
+		if (!userToken && pathname !== '/login' && pathname !== '/registration') {
+			navigate('/login');
+		}
+		if (pathname === '/') {
+			if (!userToken) {
+				navigate('/login');
+			} else {
+				navigate('/courses');
+			}
+		}
+	});
```

Comment:
Right now your hook runs every render of the App which is not optimal since you're having IO operations inside (reading from localstorage might be expensive on slow devices)

Use useEffect dependencies, I guess you only need to check pathname if it changes